File: instrument-booking/frontend/src/api/notifications.ts
import client from './client'

export interface NotificationRead {
  id: string
  type: string
  title: string
  message: string
  is_read: boolean
  created_at: string
}

export async function getNotifications(): Promise<NotificationRead[]> {
  const res = await client.get('/notifications')
  return res.data
}

export async function getUnreadCount(): Promise<number> {
  const res = await client.get('/notifications/unread-count')
  return res.data.count
}

export async function markRead(id: string): Promise<void> {
  await client.put(`/notifications/${id}/read`)
}

export async function markAllRead(): Promise<void> {
  await client.put('/notifications/read-all')
}

File: instrument-booking/backend/app/api/v1/notifications.py
import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.deps import get_current_user
from app.models.user import User
from app.schemas.common import MessageResponse
from app.schemas.notification import NotificationRead
from app.services import notification_service

router = APIRouter(prefix="/notifications", tags=["通知"])


@router.get("", response_model=list[NotificationRead])
async def list_notifications(
    page: int = Query(1, ge=1),
    per_page: int = Query(20, ge=1, le=100),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    items, _ = await notification_service.get_user_notifications(db, current_user.id, page, per_page)
    return items


@router.get("/unread-count")
async def unread_count(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    count = await notification_service.get_unread_count(db, current_user.id)
    return {"count": count}


@router.put("/{notification_id}/read", response_model=MessageResponse)
async def mark_read(
    notification_id: uuid.UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    await notification_service.mark_read(db, notification_id, current_user.id)
    return MessageResponse(message="已标记为已读")


@router.put("/read-all", response_model=MessageResponse)
async def mark_all_read(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    await notification_service.mark_all_read(db, current_user.id)
    return MessageResponse(message="全部标记为已读")
